src/System.Text.RegularExpressions/tests/GroupCollectionReadOnlyDictionaryTests.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    class GroupCollectionReadOnlyDictionaryTests
+    {
+        [Fact]
+        public static void IReadOnlyDictionary_TryGetValueSuccess()
+        {
+            Regex regex = new Regex(@"(?<A1>a*)(?<A2>b*)(?<A3>c*)");
+            Match match = regex.Match("aaabbccccccccccaaaabc");
+
+            var groups = match.Groups;
+
+            if (groups.TryGetValue("A1", out Group value))
+            {
+                Assert.Equal("aaa", value?.Value);
+            }
+            else
+            {
+                Assert.True(false, "Value should exist");
+            }
+        }
+
+        [Fact]
+        public static void IReadOnlyDictionary_TryGetValue_DoesntExist()
+        {
+            Regex regex = new Regex(@"(?<A1>a*)(?<A2>b*)(?<A3>c*)");
+            Match match = regex.Match("aaabbccccccccccaaaabc");
+
+            var groups = match.Groups;
+
+            if (groups.TryGetValue("A1", out Group value))
+            {
+                Assert.True(false, "Value should not exist");
+            }
+            else
+            {
+                Assert.Null(value);
+            }
+        }
+
+        [Fact]
+        public static void IReadOnlyDictionary_TryGetValue_NoMatch()
+        {
+            Regex regex = new Regex(@"(?<A1>a*)(?<A2>b*)(?<A3>c*)");
+            Match match = regex.Match("def");
+
+            var groups = match.Groups;
+
+            if (groups.TryGetValue("A1", out Group value))
+            {
+                Assert.Null(value);
+            }
+            else
+            {
+                Assert.True(false, "Value should exist");
+            }
+        }
+
+        [Fact]
+        public static void IReadOnlyDictionary_TryGetValue_Number()
+        {
+            Regex regex = new Regex(@"(?<A1>a*)(?<A2>b*)(?<A3>c*)");
+            Match match = regex.Match("def");
+
+            var groups = match.Groups;
+
+            if (groups.TryGetValue("0", out Group value))
+            {
+                Assert.Null(value);
+            }
+            else
+            {
+                Assert.True(false, "Value should exist");
```

Comment:
Bit odd: maybe write something like
```cs
Assert.True(groups.TryGetValue("O", out Group value));
Assert.Null(value)
```

Reply:
Well of course that would be possible too. Why do you find it a bit odd though. I think they are rather equivalent. (I picked this because the trygetvalue inside an if would be how I would be using this code elsewhere)